android/tangram/jni/platform_android.cpp

```diff
@@ -266,5 +276,16 @@ void setCurrentThreadPriority(int priority) {
     setpriority(PRIO_PROCESS, tid, priority);
 }
 
+void featureSelectionCallback(JNIEnv* jniEnv, const std::vector<std::shared_ptr<Tangram::Properties>>& items) {
+
+    jlong jresult = 0;
+    jclass propertiesClass = jniEnv->FindClass("com/mapzen/tangram/Properties");
+
+    jresult = reinterpret_cast<jlong>(new std::shared_ptr<Tangram::Properties >(items[0]));
```

Comment:
The other code just changed to also only return one feature for now. Passing a vector of properties to java is easy with swig, basically:

    jclass propertiesClass = jniEnv->FindClass("com/mapzen/tangram/Features");
    jresult = reinterpret_cast<jlong>(new std::shared_ptr<Tangram::Features >(items));

Plus %template(Features) std::vector<std::shared_ptr<Tangram::Feature>>;

I would suggest to concentrate on getting it working for the eraser-map use, and add more advanced things later.